docs/gc.md

```diff
@@ -130,17 +130,47 @@ There are a number of advantages to using arraylets.
 <i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** Despite the general advantage of using arraylets, they can slow down processing when the Java Native Interface (JNI) is being used. The JNI provides flexibility by enabling Java programs to call native code; for example C or C++, and if direct addressability to the inside of an object is needed, a JNI critical section can be used. However, that requires the object to be in a contiguous region of memory, or at least _appear_ to be so. The JNI therefore creates a temporary contiguous array that is the same size as the original array and copies everything, element by element, to the temporary array. After the JNI critical section is finished, everything is copied from the temporary array back to the arraylet, element by element.
 
 
+### `optavgpause` and `optthruput` policies
+
+Both of these policies use the global collector to manage a flat heap comprised of a single region. The global collector runs to mark and sweep the heap when the heap is exhausted (`optthruput`), or is started preemptively so that the cycle finishes before heap is completely exhausted (`optavgpause`). Additionally, if the heap becomes fragmented, the global collector compacts the heap.
+
+You are likely to find that in many situations, the default [`gencon` policy](#gencon-policy-default) offers better performance than these two policies.
+
+The primary advantage of `optthruput` is that it is free of any object access barriers (unlike `gencon` that has a write barrier to support concurrent mark, and a generational write barrier). In some workloads, the cost of those barriers might be high enough to outweigh the benefits of other policies.
+
+The `optavgpause` policy similarly lacks a generational write barrier, though it does have a write barrier to support concurrent mark. But worklaods that frequently change large and old reference arrays, for example, might have a costly generational barrier and would therefore benefit from this policy.
+
+An additional potential problem with generation GCs is that it splits heap so that a very large object in worst case might not fit in either of the 2 areas, while there is enough total free memory, or in better case, allocation may succeed but require additional GC cycles and copying of objects (relative to non-generational GC) to accommodate large enough contiguous free space.
```

Comment:
?? Neither of these policies are generational, so this is, I guess a reason for choosing flat heaps over generational. However, I think we need to break this down a bit to make it more consumable.

By using a *flat* heap, these policies avoid potential issues with very large objects. With a generational policy, the heap is divided into smaller spaces in order to manage generations of objects. For `gencon` this is the *nursery* area and for `balanced` this is the *eden* region. Although there might be sufficient free space on the overall Java heap for a very large object, it might not fit into these spaces. If the allocator does succeeds in allocating a very large object, further GC cycles might be required to create enough contiguous free space.

So for me, this becomes a "When to use" (see my TOC suggestions for laying out the policies) .... i.e. If your application requires very large objects that persist on the Java heap ..... 

@amicic - is this what you intended here?

Reply:
Correct, these are not generational, so the intentions was to enlist potential weakness of generational GCs, more specifically Gencon (object access barrier and heap segregation) as a justification to consider using Flat heap based ones.

I don't know if it's wise to mention here Balanced/Eden etc.  In Balanced, the problem of large object allocation that does not fit into a region is resolved with arraylets. BTW, Eden is actually a set of regions, not a single region.

Overall the text good to me (probably because it's based on my suggestions :) )

minor mistakes:
- problem with generational GCs (not generation)
- worklaods  -> workloads